core/src/main/java/org/jruby/util/Inspector.java

```diff
@@ -0,0 +1,56 @@
+package org.jruby.util;
+
+import org.jcodings.specific.USASCIIEncoding;
+import org.jruby.Ruby;
+import org.jruby.RubyModule;
+import org.jruby.RubyString;
+import org.jruby.runtime.ThreadContext;
+
+import static org.jruby.util.io.EncodingUtils.encStrBufCat;
+
+public class Inspector {
+
+    public static final ByteList EMPTY_ARRAY_BYTES = new ByteList(new byte[] { '[',']' }, USASCIIEncoding.INSTANCE, false);
+    public static final ByteList RECURSIVE_ARRAY_BYTES = new ByteList(new byte[] { '[','.','.','.',']' }, USASCIIEncoding.INSTANCE, false);
+
+    public static final ByteList EMPTY_HASH_BYTES = new ByteList(new byte[] { '{','}' }, USASCIIEncoding.INSTANCE, false);
+    public static final ByteList RECURSIVE_HASH_BYTES = new ByteList(new byte[] { '{','.','.','.','}' }, USASCIIEncoding.INSTANCE, false);
```

Comment:
I know "BYTELIST" is longer than "BYTES"... maybe we can use "BL" to indicate that these are more than just the bytes?  Just a minor nitpick.